### ISSUE TITLE
hyperparameters

### ISSUE THREAD

**Issue by XLLXO** [quotation2520/PG-RCNN#10]
Hello Author:
I would like to ask you a question
My training set and validation set are assigned 5:5
The training set is 3712 samples and the validation set is 3769 samples.
I first train the model using the training set, and after getting better results on the validation set
After getting better results on the validation set, I directly tested the model with the weights and submitted it to KITTI official test server and got result A.

Then I train the model using the training set + validation set (7481 samples in total) with all the same hyperparameters as above.
After training, I tested the model with the obtained weights, and submitted it to KITTI official test server, and got the result B.
However, in the Pedestrian (3D Detection) category, the detection accuracy of B is lower than A by 1-2 points.

I reduced the training epoch by half but it didn't solve the problem.

What is the reason for this?
How should I adjust the hyperparameters to solve this problem?

I am looking forward to your reply!
Thank you!

**Comment by quotation2520** [comment]
Hi, achieving a good performance on a test set is a difficult problem.
Even I don't know the best way to finetune model for test performance, but I can tell you this:

Using all the available labeled data for training does NOT necessarily guarantee the best performance.
You should have probably noticed that the last epoch is often not the best epoch.
If you used all the train+val data for training, how did you find the best epoch?
Your model may have **overfitted** to the training data.
That is why you need a validation set, to examine if your trained model perform well on unseen data too.
If the validation set has similar data distribution with the test set, than reaching a good performance on the val set would be helpful.

You need a larger training data for better generalization capability, but still you need a isolated validation set for finetuning and model selection. 

Some papers use 80%-20% data split, others use different split. Ensembling is often introduced.
There are many minor tricks that researchers don't say on papers.
For instance, I examined the qualitative results on the test set to finetune a nice post-processing threshold.

I hope you find a good procedure of your own.
Good luck.

**Comment by XLLXO** [comment]
> Hi, achieving a good performance on a test set is a difficult problem. Even I don't know the best way to finetune model for test performance, but I can tell you this:
> 
> Using all the available labeled data for training does NOT necessarily guarantee the best performance. You should have probably noticed that the last epoch is often not the best epoch. If you used all the train+val data for training, how did you find the best epoch? Your model may have **overfitted** to the training data. That is why you need a validation set, to examine if your trained model perform well on unseen data too. If the validation set has similar data distribution with the test set, than reaching a good performance on the val set would be helpful.
> 
> You need a larger training data for better generalization capability, but still you need a isolated validation set for finetuning and model selection.
> 
> Some papers use 80%-20% data split, others use different split. Ensembling is often introduced. There are many minor tricks that researchers don't say on papers. For instance, I examined the qualitative results on the test set to finetune a nice post-processing threshold.
> 
> I hope you find a good procedure of your own. Good luck.

Thank you very much for your reply!
I would like to ask one more question:
If I use 80%-20% data split or 90%-10% data split
Can you tell me how to select the training set and validation set?
Do you have a txt file with the training and validation sets already split? Can you share it?
Here is my email: 2102012@stu.neu.edu.cn

**Comment by quotation2520** [comment]
Hi, sorry I checked the added comment too late.

I just randomly split 80/20 of train+val data without any sophiscated consideration.
Here are the splits, but I'm not sure it's a good one.

[train_8020.txt](https://github.com/quotation2520/PG-RCNN/files/14833999/train_8020.txt)
[val_8020.txt](https://github.com/quotation2520/PG-RCNN/files/14834000/val_8020.txt)

Thanks!
